### PR TITLE
add unitttest for mac on ci after some untest being disable

### PR DESCRIPTION
**Pull request by JiabinYang**

### CODE REVIEW

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 1]
```diff
@@ -17,12 +17,16 @@ function(download_model_and_data install_dir model_name data_name)
...
-         --infer_data=${RNN1_INSTALL_DIR}/data.txt)
+# TODO: fix this test on MACOS
+message(WARNING "These tests has been disabled in OSX before being fixed: \n test_analyzer_rnn1")
+if(NOT APPLE)
+    set(RNN1_INSTALL_DIR "${INFERENCE_DEMO_INSTALL_DIR}/rnn1")
+    download_model_and_data(${RNN1_INSTALL_DIR} "rnn1%2Fmodel.tar.gz" "rnn1%2Fdata.txt.tar.gz")
+    inference_analysis_test(test_analyzer_rnn1 SRCS analyzer_rnn1_tester.cc 
+        EXTRA_DEPS ${INFERENCE_EXTRA_DEPS}
+        ARGS --infer_model=${RNN1_INSTALL_DIR}/model
+            --infer_data=${RNN1_INSTALL_DIR}/data.txt)
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/CMakeLists.txt:28]: 请问这个单侧在mac上报什么错呢？

Comment (JiabinYang) [reply paddle/fluid/inference/tests/api/CMakeLists.txt:28]: 这个就像您那样修复就可以啦，这个PR就直接用了您的修复

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -67,26 +67,44 @@ function cmake_gen() {
...
-            export PATH=/opt/python/cp27-cp27mu/bin/:${PATH}
-            PYTHON_FLAGS="-DPYTHON_EXECUTABLE:FILEPATH=/opt/python/cp27-cp27mu/bin/python
-        -DPYTHON_INCLUDE_DIR:PATH=/opt/python/cp27-cp27mu/include/python2.7
-        -DPYTHON_LIBRARIES:FILEPATH=/opt/_internal/cpython-2.7.11-ucs4/lib/libpython2.7.so"
-        elif [ "$1" == "cp35-cp35m" ]; then
-            export LD_LIBRARY_PATH=/opt/_internal/cpython-3.5.1/lib/:${LD_LIBRARY_PATH}
-            export PATH=/opt/_internal/cpython-3.5.1/bin/:${PATH}
-            export PYTHON_FLAGS="-DPYTHON_EXECUTABLE:FILEPATH=/opt/_internal/cpython-3.5.1/bin/python3
+    SYSTEM=`uname -s`
+    if [ SYSTEM == "Darwin" ]; then
```
Comment (typhoonzero) [inline paddle/scripts/paddle_build.sh]: `if [ "$SYSTEM" == "Darwin" ]; then`

File: paddle/scripts/paddle_build.sh [hunk 3]
```diff
@@ -67,26 +67,44 @@ function cmake_gen() {
...
+        if [ "$1" == "cp27-cp27m" || "$1" == "" ]; then
+            echo "using python abi: $1"
+            if [ -d "/Library/Frameworks/Python.framework/Versions/2.7" ]; then
+                export LD_LIBRARY_PATH=/Library/Frameworks/Python.framework/Versions/2.7
+                export DYLD_LIBRARY_PATH=/Library/Frameworks/Python.framework/Versions/2.7
+                export PATH=/Library/Frameworks/Python.framework/Versions/2.7/bin/:{PATH}
+                PYTHON_FLAGS="-DPYTHON_EXECUTABLE:FILEPATH=/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7
+            -DPYTHON_INCLUDE_DIR:PATH=/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7
+            -DPYTHON_LIBRARIES:FILEPATH=/Library/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7.dylib"
```
Comment (typhoonzero) [inline paddle/scripts/paddle_build.sh]: `should use -DPYTHON_LIBRARY` for mac in my case.

Comment (typhoonzero) [reply paddle/scripts/paddle_build.sh]: should test if those files exsist.

**Review by guochaorong — DISMISSED** [review]
LGTM

File: paddle/scripts/paddle_build.sh (hunk 3 above)
Comment (JiabinYang) [reply paddle/scripts/paddle_build.sh]: thx, and fixed

**Review by typhoonzero — APPROVED** [review]
LGTM

**Pull request merged**